metaflow/plugins/kfp/kfp.py

```diff
@@ -105,6 +106,56 @@ def _get_retries(node):
 
         return max_user_code_retries, max_user_code_retries + max_error_retries
 
+    @staticmethod
+    def _get_resource_requirements(node):
+        """
+        Get resource request or limit for a Metaflow step (node) set by @resources decorator.
+
+        Supported parameters: 'cpu', 'cpu_limit', 'gpu', 'gpu_vendor', 'memory', 'memory_limit'
+        Keys with no suffix set resource request (minimum);
+        keys with 'limit' suffix sets resource limit (maximum).
+
+        Eventually resource request and limits link back to kubernetes, see
+        https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+
+        Default unit for memory is megabyte, aligning with existing resource decorator usage.
+
+        Example using resource decorator:
+            @resource(cpu=0.5, cpu_limit=2, gpu=1, memory=300)
+            @step
+            def my_kfp_step(): ...
+        """
+
+        resource = dict()
+        for deco in node.decorators:
+            if isinstance(deco, ResourcesDecorator):
+                for attr_key, attr_value in deco.attributes.items():
+                    curr_attr_value = resource.get(attr_key)
+
+                    if attr_value is not None:
+                        if curr_attr_value is not None:
+                            # Resource decorator should only be used once to avoid confusion
+                            # This only happens when user statically define same resource
+                            # requirement with more than one resource decorator for a single step.
+                            # Decorators attached during runtime doesn't fall into this category
+                            # for being dropped by metaflow.decorators._attach_decorators_to_step.
+                            raise ValueError(
```

Comment:
This is already being handled here - https://github.com/Netflix/metaflow/blob/master/metaflow/decorators.py#L371